pkg/trace/api/telemetry.go

```diff
@@ -0,0 +1,136 @@
+package api
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	stdlog "log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/info"
+	"github.com/DataDog/datadog-agent/pkg/trace/logutil"
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// telemetryMultiTransport sends HTTP requests to multiple targets using an
+// underlying http.RoundTripper. API keys are set separately for each target.
+// The target hostname
+// When multiple endpoints are in use the response from the main endpoint
+// is proxied back to the client, while for all aditional endpoints the
+// response is discarded. There is no de-duplication done between endpoint
+// hosts or api keys.
+//
+// Could be extended in the future to allow supporting more product endpoints
+// by simply parametrizing metric tags, and logger names
+type telemetryMultiTransport struct {
+	Transport http.RoundTripper
+	Endpoints []*config.Endpoint
+}
+
+// telemetryProxyHandler parses returns a new HTTP handler which will proxy requests to the configured intakes.
+// If the main intake URL can not be computed because of config, the returned handler will always
+// return http.StatusInternalServerError along with a clarification.
+func (r *HTTPReceiver) telemetryProxyHandler() http.Handler {
+	// extract and validate Hostnames from configured endpoints
+	var endpoints []*config.Endpoint
+	for _, endpoint := range r.conf.TelemetryConfig.Endpoints {
+		u, err := url.Parse(endpoint.Host)
+		if err != nil {
+			log.Errorf("Error parsing apm_config.telemetry endpoint: %s", endpoint.Host)
```

Comment:
```suggestion
			log.Errorf("Error parsing apm_config.telemetry endpoint %q: %v", endpoint.Host, err)
```
Maybe the error ends up being useful?